numaprom/clients/sentinel.py

```diff
@@ -0,0 +1,48 @@
+from typing import Optional
+
+from redis.backoff import ExponentialBackoff
+from redis.client import Redis
+from redis.exceptions import RedisClusterException, RedisError
+from redis.retry import Retry
+from redis.sentinel import Sentinel, MasterNotFoundError
+
+from numaprom import get_logger
+
+_LOGGER = get_logger(__name__)
+SENTINEL_MASTER_CLIENT: Optional[Redis] = None
+
+
+def get_redis_client(
+    host: str, port: int, password: str, mastername: str, recreate: bool = False
+) -> Redis:
+    """
+    Return a master redis client for sentinel connections, with retry.
+    """
+    global SENTINEL_MASTER_CLIENT
+
+    if not recreate and SENTINEL_MASTER_CLIENT:
+        return SENTINEL_MASTER_CLIENT
+
+    retry = Retry(
+        ExponentialBackoff(cap=2, base=1),
```

Comment:
With this setting, we are in effect not doing exponential backoff.

https://redis-py.readthedocs.io/en/stable/_modules/redis/backoff.html#ExponentialBackoff

Reply:
You want the numbers to be smaller?